src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionMultilineStringArgsInspection.kt

```diff
@@ -40,20 +42,32 @@ class SmkSectionMultilineStringArgsInspection : SnakemakeInspection() {
                 }
             }
         }
+    }
 
-        private fun checkRecursive(expr: PyExpression?): Boolean {
-            return when (expr) {
-                is PyBinaryExpression -> checkRecursive(expr.leftExpression)
-                        || checkRecursive(expr.rightExpression)
-                is PyParenthesizedExpression -> checkRecursive(expr.containedExpression)
-                is PyStringLiteralExpression -> expr.decodedFragments.size > 1
-                        && expr.stringElements.any { x ->
-                    x.nextSibling is PsiWhiteSpace && x.nextSibling.textContains('\n')
-                }
-                else -> false
-            }
+    override fun getDisplayName(): String = SnakemakeBundle.message("INSP.NAME.section.multiline.string.args")
+}
+
+class SmkMultilineStringArgsInspectionVisitor : PyElementVisitor() {
+
+    var warning: Boolean = false
+        private set
+
+    override fun visitPyBinaryExpression(node: PyBinaryExpression) {
+        node.leftExpression.accept(this)
```

Comment:
Please check PyRecursiveElementVisitor for a more elegant and abstract solution for visiting children

Reply:
The idea was to check impl of PyRecursiveElementVisitor, not use it. Not inspection is more aggressive and looks recursively even in method calls, etc. E.g.:

![image](https://user-images.githubusercontent.com/72933/87432252-6b39e280-c5f0-11ea-8a0d-d898e3f2e9bc.png)
I don't think that warning for `foo` call here is required. Please add tests for such case.

Reply:
Done